ExampleMod/Content/Items/CustomItemSets.cs

```diff
@@ -0,0 +1,63 @@
+	[ReinitializeDuringResizeArrays]
+	public static class CustomItemSets
+	{
+		// Custom item set example. This will behave the same as any other ItemID.Sets array.
+		public const string FlamingWeaponCustomSetKey = "FlamingWeapon"; // This string MUST be consistent between mods working together.
+
+		// To create a custom item set, we use the ItemID.Sets.Factory.CreateNamedXSet method and provide a string key.
+		// We can also pass in any initial data, in this case we are indicating that FieryGreatsword and ExampleSword should be true in this set. Note that it is also possible to set these set values in SetStaticDefaults instead, which is more typical. See ExampleFlail.cs for an example.
+		// This method also exposes the set for other mods to access via this key. The key and default value must be consistent with other mods.
+		public static bool[] FlamingWeapon = ItemID.Sets.Factory.CreateNamedBoolSet(FlamingWeaponCustomSetKey, false, ItemID.FieryGreatsword, ModContent.ItemType<ExampleSword>());
+		// Note that by using the ReinitializeDuringResizeArrays approach, ModContent.ItemType<ExampleSword>() is a valid input since modded content IDs will be assigned and retrievable during the reinitialization. Without ReinitializeDuringResizeArrays the code will incorrectly use 0 as the value of ModContent.ItemType<ExampleSword>() because modded IDs haven't been assigned yet.
+
+		// If a set is specific to content in your mod, consider using the CreateNamedBoolSet(string modName, string key, ...) overload instead to avoid any potential conflicts with another mod that happens to use that same key for unrelated purposes.
+	}
+
+	public class CustomItemSetsSystem : ModSystem
+	{
+		public override void ResizeArrays() {
+			// ResizeArrays is the earliest method called after all content has loaded and have been assigned ID values.
+			// This is where methods such as SetFactory.CreateNamedBoolSet should be called if not using the ReinitializeDuringResizeArrays attribute to do this automatically with a field initializer.
+
+			// For example, we could put "CustomItemSets.FlamingWeapon = ItemID.Sets.Factory.CreateNamedBoolSet(FlamingWeaponCustomSetKey, false, ItemID.FieryGreatsword);" here instead of in the CustomItemSets class field initializers.
+		}
+
+		public override void SetStaticDefaults() {
+			// We can further edit the data sets here. These changes will still be consistent between all mods accessing this set since the object reference is shared.
+			CustomItemSets.FlamingWeapon[ItemID.FireWhip] = true;
+			CustomItemSets.FlamingWeapon[ItemID.HelFire] = true;
+
+			// For example, we can add user-defined items to CustomItemSets.FlamingWeapon
+			// Note that this example would need more code to work correctly for config values changed in-game, this is just showing a potential use-case.
```

Comment:
config values changed ingame for something like this should have a `NeedsReload`. It's likely that custom sets will be used in `AppliesToEntity` methods which are frozen after loading completes.

Reply:
(Will be in a separate PR since there are more issues with ItemDefinition/ItemID.Search)